server/client.go

```diff
@@ -1708,22 +1716,25 @@ func (c *client) generateClientInfoJSON(info Info) []byte {
 
 func (c *client) sendErr(err string) {
 	c.mu.Lock()
-	c.traceOutOp("-ERR", []byte(err))
+	c.traceOutOpIfOk("-ERR", []byte(err))
 	c.enqueueProto([]byte(fmt.Sprintf(errProto, err)))
 	c.mu.Unlock()
 }
 
 func (c *client) sendOK() {
 	c.mu.Lock()
-	c.traceOutOp("OK", nil)
+	c.traceOutOpIfOk("OK", nil)
 	c.enqueueProto([]byte(okProto))
 	c.pcd[c] = needFlush
 	c.mu.Unlock()
 }
 
 func (c *client) processPing() {
 	c.mu.Lock()
-	c.traceInOp("PING", nil)
+	if c.trace {
```

Comment:
Tried to avoid having multiple versions of tracing functions.
I added traceOutOpIfOk, but NO traceInOpIfOk or traceOpIfOk. 
You pin pointed 3 instances where they could have been used.

I'll see how much mileage i get out of pushing trace into the caller. Hopefully I can get rid of traceOutOpIfOk.